pkg/installmanager/installmanager.go

```diff
@@ -530,41 +527,30 @@ func (m *InstallManager) cleanupFailedInstall(cd *hivev1.ClusterDeployment, prov
 }
 
 func cleanupFailedProvision(dynClient client.Client, cd *hivev1.ClusterDeployment, infraID string, logger log.FieldLogger) error {
+	var uninstaller providers.Destroyer
 	switch {
 	case cd.Spec.Platform.AWS != nil:
 		// run the uninstaller to clean up any cloud resources previously created
 		filters := []aws.Filter{
 			{kubernetesKeyPrefix + infraID: "owned"},
 		}
-		uninstaller := &aws.ClusterUninstaller{
+		uninstaller = &aws.ClusterUninstaller{
 			Filters: filters,
 			Region:  cd.Spec.Platform.AWS.Region,
 			Logger:  logger,
 		}
-
-		if err := uninstaller.Run(); err != nil {
-			return err
-		}
-
-		// If we're managing DNS for this cluster, lookup the DNSZone and cleanup
-		// any leftover A records that may have leaked due to
-		// https://jira.coreos.com/browse/CORS-1195.
-		return cleanupDNSZone(dynClient, cd, logger)
 	case cd.Spec.Platform.Azure != nil:
-		uninstaller := &azure.ClusterUninstaller{}
-		uninstaller.Logger = logger
-		session, err := azuresession.GetSession(installertypesazure.PublicCloud)
-		if err != nil {
-			return err
+		metadata := &installertypes.ClusterMetadata{
+			InfraID: infraID,
+			ClusterPlatformMetadata: installertypes.ClusterPlatformMetadata{
+				Azure: &installertypesazure.Metadata{
+					CloudName: installertypesazure.PublicCloud,
```

Comment:
will it always be type PublicCloud?

Reply:
Yes, until we do https://issues.redhat.com/browse/CO-1005.